test/bun.js/oniguruma-regexp.test.ts

```diff
@@ -2,9 +2,156 @@ import { OnigurumaRegExp } from "bun";
 import { expect, it, test } from "bun:test";
 import { gc as gcTrace } from "./gc";
 
+it("repeated match and exec calls", () => {
+  for (let i = 0; i < 20000; i++) {
+    let r1 = new OnigurumaRegExp("//.+?/[^?]+", "sg");
+    let r2 = new RegExp("//.+?/[^?]+", "sg");
+    let s1 = "https://dylan-conway.com/profile";
+    expect(s1.match(r1)[0] === s1.match(r2)[0]).toBe(true);
+    expect(r1.exec(s1)[0] === r2.exec(s1)[0]).toBe(true);
+  }
+});
+
+it("repeated match and exec calls no global flag", () => {
+  for (let i = 0; i < 20000; i++) {
+    let r1 = new OnigurumaRegExp("//.+?/[^?]+", "s");
+    let r2 = new RegExp("//.+?/[^?]+", "s");
+    let s1 = "https://dylan-conway.com/profile";
+    expect(r1.exec(s1)[0] === r2.exec(s1)[0]).toBe(true);
+    expect(s1.match(r1)[0] === s1.match(r2)[0]).toBe(true);
+  }
+});
+
+const rb1 = new OnigurumaRegExp("//.+?/[^?]+", "s");
+const rb2 = new RegExp("//.+?/[^?]+", "s");
+it("repeated match calls with global regex without global flag", () => {
+  for (let i = 0; i < 20000; i++) {
+    let s1 = "https://dylan-conway.com/profile";
+    expect(rb1.exec(s1)[0] === rb2.exec(s1)[0]).toBe(true);
+    expect(s1.match(rb1)[0] === s1.match(rb2)[0]).toBe(true);
+  }
+});
+
 it("escaped characters in character classes", () => {
+  expect(
+    new RegExp("[a-z]").exec("a")[0] ===
+      new OnigurumaRegExp("[a-z]").exec("a")[0]
+  ).toBe(true);
+
+  expect(
+    new RegExp("[a-z]").exec("b")[0] ===
+      new OnigurumaRegExp("[a-z]").exec("b")[0]
+  ).toBe(true);
```

Comment:
Let's add some `Bun.gc(true)` in here to make sure freeing these doesn't crash

Reply:
Added some random ones. I tried adding some inside the for loops as well but didn't keep them because it's slow.